src/com/xilinx/rapidwright/interchange/DeviceResourcesWriter.java

```diff
@@ -321,6 +321,8 @@ public static void writeDeviceResourcesFile(String part, Device device, CodePerf
 
         removeUnusedMacros(macros, prims);
 
+        macros = netlist.copyLibraryAndSubCells(macros);
```

Comment:
```suggestion
        // Perform a deep copy because macro cells (which were shallow copied before)
        // must now instantiate primitives from our primitives library
        macros = netlist.copyLibraryAndSubCells(macros);
```